text/chapter1.md

```diff
@@ -36,6 +36,18 @@ Functions enable a simple form of abstraction which can yield great productivity
 
 PureScript is a programming language which aims to address these issues. It features lightweight syntax, which allows us to write very expressive code which is still clear and readable. It uses a rich type system to support powerful abstractions. It also generates fast, understandable code, which is important when interoperating with JavaScript, or other languages which compile to JavaScript. All in all, I hope to convince you that PureScript strikes a very practical balance between the theoretical power of purely functional programming, and the fast-and-loose programming style of JavaScript.
 
+### About TypeScript and Elm
+
+PureScript, TypeScript, and Elm, are often compared, so it's important to talk about some major differences between them.
+
+On TypeScript side, the type system is intentionally unsound by design to enable full JavaScript compatibility, and PureScript core features around the type system can be reproduce, but only partially and with a lot of boilerplate (like pattern matching on ADTs).
```

Comment:
I'd also like to add some additional emphasis here on the very pointed difference between TypeScript's type checker and PureScript's type system.  Feel free to take my following text and edit or make use of any of it as you feel is appropriate:  TypeScript, by its nature, its design, and the constraints it is burdened with due to the design and quirks of JavaScript, *cannot make any guarantees about the program*, whether in correctness, soundness, or safety.  Indeed, this is intentional as TypeScript is intended to be an enhancement of JavaScript, and not a replacement; but this difference is absolutely fundamental between what PureScript is and what TypeScript is.  TypeScript's type inference is also extremely limited in what it can do, partly because a great many cases can be interpreted in just too many different ways for the tool to confidently decide what the real intention is. 

PureScript's type system, on the other hand, *does make those guarantees*.  Because of this and the completeness of PureScript's type system, it has an amazing and intelligent type inference which can dig through layer after layer in your program and choose a type based on whether all the operations conform to something specific. There are even some programs which can technically be written without a single explicit type definition!  (Not recommended).

However, the trade-off is that PureScript has to throw out all JavaScript semantics and design.  The only role JS serves is as a compile target.  PureScript is a vastly different language, and to work with it, you at least have to be brave enough to try something that is so very different to perhaps anything you have ever known; however, we believe this to be a very worthwhile trade.

Reply:
I'm uncomfortable with this wording, especially the "cannot make any guarantees" part. To me that feels like a much stronger claim about the efficacy of PureScript versus TypeScript than I can imagine being justifiable. I understand that TypeScript's type system is known to be unsound whereas we'd probably consider it a bug if it turned out that PureScript's type system wasn't sound (note: nobody has proved that PureScript's type system is sound), but beyond that I don't see how you can objectively arrive at the conclusion that TypeScript cannot make any guarantees about any program whereas PureScript can.

> However, the trade-off is that PureScript has to throw out all JavaScript semantics and design.

This is not true; PureScript inherits a lot of JavaScript's semantics and design. PureScript's semantics follow JavaScript's in many ways precisely because it was intended to interop with JS without too much hassle. For example, PureScript is strict, like JS, and unlike Haskell. Also, JS's event loop model is explicit in the Effect/Aff distinction and you have to worry about that as a PureScript developer. You also have to worry about your stack size occasionally, something which is generally not an issue in Haskell.

Reply:
Fair enough, I'm probably not knowledgeable enough at this stage to offer any useful contributions. Thank you for the feedback though.